### PR TITLE
Use less things from scalaz

### PR DESCRIPTION
**Pull request by alexarchambault**
This PR is a first step towards fixing https://github.com/coursier/coursier/issues/175. It substitutes  `scala.Either` to `scalaz.\/` everywhere, and a custom `coursier.util.EitherT` to `scalaz.EitherT`.

### CODE REVIEW

**Comment by leonardehrenfried** [comment]
I think this is great step into the right direction!

It seems that the tut compilation failed.

**Comment by alexarchambault** [comment]
I'll mention in the README that it's about the SNAPSHOT version of coursier, and a link to the latest stable release, before updating it.

File: project/Mima.scala [hunk 1]
```diff
@@ -10,25 +10,6 @@ object Mima {
...
-    "1.0.0-RC10",
-    "1.0.0-RC11",
-    "1.0.0-RC12",
-    "1.0.0-RC12-1",
-    "1.0.0-RC13",
-    "1.0.0-RC14",
-    "1.0.0",
-    "1.0.1-M1",
-    "1.0.1",
-    "1.0.2",
```
Comment (wisechengyi) [inline project/Mima.scala]: woot! looking forward to remove those deprecated methods as well!

**Comment by alexarchambault** [comment]
Waiting a bit before merging to give https://github.com/coursier/coursier/pull/774 the priority (small chance of conflicts)

**Pull request merged**